### DISCUSSION TITLE
Conditionally required logic or separate the required logic of admin and field.

### DISCUSSION THREAD

**Discussion by Stupidism** [payloadcms/payload#2338] [Feature Requests & Ideas]
For example 
Some fields are required if type is external

![image](https://user-images.githubusercontent.com/5801015/225835771-dc4ed46a-6d20-445d-a69e-a035cfc354a8.png)

The other field article is required if type is internal

<img width="683" alt="image" src="https://user-images.githubusercontent.com/5801015/225835851-dbc6f18d-30aa-444c-8e4b-9e4f0d94269e.png">

**Comment by ssyberg** [comment]
👍🏼 would love to see this, also wondering if there's an existing ticket for this?

**Comment by hjbugajski** [comment]
@Stupidism @ssyberg Hello! This is possible, you just need to add `required: true` on the field that’s conditionally rendered. The validation only occurs when the field is rendered, so setting required on that field works as expected.

```javascript
{
  name: 'url',
  label: 'External Link',
  type: 'text',
  required: true,
  admin: {
    condition: (_, siblingData) => siblingData?.type === 'external'
  }
}
```

**Reply by ssyberg** [reply]
This doesn't quite get us there, some fields are still shown but become optional depending on the value of sibling data. This solve is a custom validator but you then lose the native admin `required` field treatment.

**Reply by hjbugajski** [reply]
I see. The original post reads as marking fields that are specific to a certain type as required, not fields that match multiple types but are only required for a subset. In that case, then a conditional required functionality is needed.

It's not ideal, but you can achieve this by having duplicate fields (one with required, one without) and conditionally render them based on the type.

<details>
<summary>Example code</summary>

```typescript
export const linkGroup: Field = {
  name: 'link',
  type: 'group',
  fields: [
    {
      name: 'text',
      type: 'text',
      required: true,
    },
    {
      name: 'type',
      type: 'radio',
      admin: {
        layout: 'horizontal',
      },
      defaultValue: 'internal',
      required: true,
      options: [
        {
          label: 'Internal',
          value: 'internal',
        },
        {
          label: 'External',
          value: 'external',
        },
      ],
    },
    {
      name: 'demoField',
      type: 'text',
      required: true,
      admin: {
        condition: (_, siblingData) => siblingData?.type === 'internal',
      },
    },
    {
      name: 'demoField',
      type: 'text',
      admin: {
        condition: (_, siblingData) => siblingData?.type === 'external',
      },
    },
  ],
};
```
</details>
